packages/sqlite_async/lib/src/native/database/connection_pool.dart

```diff
@@ -232,6 +239,66 @@ class SqliteConnectionPool with SqliteQueries implements SqliteConnection {
       await connection.refreshSchema();
     }
   }
+
+  Future<T> withAllConnections<T>(
+      Future<T> Function(
+              SqliteWriteContext writer, List<SqliteReadContext> readers)
+          block) async {
+    try {
+      _runningWithAllConnections = true;
+
+      final blockCompleter = Completer<T>();
+      final (write, reads) = await _lockAllConns<T>(blockCompleter);
+
+    try {
+      final res = await block(write, reads);
+      blockCompleter.complete(res);
+      return res;
+    } catch (e, st) {
+        blockCompleter.completeError(e, st);
+        rethrow;
+      }
+    } finally {
+      _runningWithAllConnections = false;
```

Comment:
I think there's a potential race when calling this method multiple times, i.e. with

1. First call sets `_runningWithAllConections = true`.
2. Second call sets `_runningWithAllConections = true`.
3. First call obtains connections and runs the callback.
4. First call is done with the callback.
5. Second call obtains connections and runs the callback.
6. First call sets `_runningWithAllConnections = false`, so a new connection could be spawned while the second `withAllConnections` call is still active?

Making this an `int` to increase / decrease should fix this.